trace/trace.go

```diff
@@ -0,0 +1,209 @@
+// package trace implement structured logging of requests
+package trace
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/mailgun/oxy/utils"
+)
+
+// TraceOption is a functional option setter
+type TraceOption func(*Tracer) error
+
+// ErrorHandler is a functional argument that sets error handler of the server
+func ErrorHandler(h utils.ErrorHandler) TraceOption {
+	return func(t *Tracer) error {
+		t.errHandler = h
+		return nil
+	}
+}
+
+// Add request headers to capture
+func RequestHeaders(headers ...string) TraceOption {
+	return func(t *Tracer) error {
+		t.reqHeaders = headers
```

Comment:
How about this?

```go
t.reqHeaders = append(t.reqHeaders, headers)
```

That way if there are multiple RequestHeaders options set, they each add request headers rather than overwriting each other. Or you could change the doc to say that RequestHeaders _sets_ the headers to capture.

Same with ResponseHeaders.